framework/src/base/FEProblem.C

```diff
@@ -1202,22 +1194,14 @@ FEProblem::addFunction(std::string type, const std::string & name, InputParamete
   for (THREAD_ID tid = 0; tid < libMesh::n_threads(); tid++)
   {
     MooseSharedPointer<Function> func = MooseSharedNamespace::static_pointer_cast<Function>(_factory.create(type, name, parameters, tid));
-    if (_functions[tid].find(name) != _functions[tid].end())
-      mooseError("Duplicate function name added to FEProblem: " << name);
```

Comment:
This is all handled by the InputParameterWarehouse, it is impossible to have any objects with the same now.

https://github.com/idaholab/moose/blob/devel/framework/src/utils/InputParameterWarehouse.C#L47